frontend/src/App.tsx

```diff
@@ -0,0 +1,152 @@
+import React from "react";
+import { Provider } from "react-redux";
+import {
+  ApolloClient,
+  ApolloProvider,
+  ApolloLink,
+  InMemoryCache,
+  concat,
+} from "@apollo/client";
+import {
+  Route,
+  BrowserRouter as Router,
+  Routes,
+  Navigate,
+} from "react-router-dom";
+import { createUploadLink } from "apollo-upload-client";
+import { ErrorResponse, onError } from "@apollo/client/link/error";
+import { ApplicationInsights } from "@microsoft/applicationinsights-web";
+import Modal from "react-modal";
+
+import App from "./app/ReportingApp";
```

Comment:
thank you for refactoring the old index.tsx file and explaining why that was needed!! 👍